codegen/src/generator/rest_json.rs

```diff
@@ -0,0 +1,144 @@
+            format!("
+                {documentation}
+                pub fn {method_name}(&mut self, input: &{input_type}) -> AwsResult<{output_type}> {{
+                    {encode_input}
+
+                    {request_uri_formatter}
+
+                    let mut request = SignedRequest::new(\"{http_method}\", \"{endpoint_prefix}\", self.region, &request_uri);
+                    request.set_content_type(\"application/x-amz-json-1.1\".to_owned());
+                    {load_payload}
+                    let mut result = request.sign_and_execute(try!(self.credentials_provider.credentials()));
+                    let status = result.status.to_u16();
+                    let mut body = String::new();
+                    result.read_to_string(&mut body).unwrap();
+                    match status {{
+                        {status_code} => {{
+                            {ok_response}
+                        }}
+                        _ => Err(parse_json_protocol_error(&body)),
+                    }}
+                }}
+                ",
+                documentation = generate_documentation(operation).unwrap_or("".to_owned()),
+                endpoint_prefix = service.metadata.endpoint_prefix,
+                http_method = operation.http.method,
+                input_type = input_type,
+                method_name = operation.name.to_snake_case(),
+                status_code = operation.http.response_code.unwrap_or(200),
+                ok_response = generate_ok_response(operation, output_type),
+                output_type = output_type,
+                request_uri_formatter = generate_uri_formatter(&operation.http.request_uri, &member_strings),
+                load_payload = generate_payload_loading_string(load_payload),
+                encode_input = generate_encoding_string(load_payload),
+            )
+        }).collect::<Vec<String>>().join("\n")
+    }
+
+    fn generate_prelude(&self) -> String {
+        "use std::io::Read;
+
+        use serde_json;
+
+        use credential::ProvideAwsCredentials;
+        use error::{AwsResult, parse_json_protocol_error};
+        use region::Region;
+        use signature::SignedRequest;
+        ".to_owned()
+    }
+
+    fn generate_struct_attributes(&self) -> String {
+        "#[derive(Debug, Default, Deserialize, Serialize)]".to_owned()
+    }
+}
+
+fn generate_encoding_string(load_payload: bool) -> String {
+    match load_payload {
+        false => "".to_owned(),
+        true => {
+            format!(
+                "let encoded = serde_json::to_string(input).unwrap();"
+            )
+        },
+    }
+ }
+
+fn generate_payload_loading_string(load_payload: bool) -> String {
+    match load_payload {
+        false => "".to_owned(),
+        true => {
+            format!(
+                "request.set_payload(Some(encoded.as_bytes()));"
+            )
+        },
+    }
+}
+
+fn generate_uri_formatter(request_uri: &str, member_strings: &Vec<String>) -> String {
```

Comment:
This section causes snake case warnings.  Elastictranscoder.rs codegen output for example:

```rust
let request_uri = format!("/2012-09-25/pipelines/{Id}/notifications" , Id = input.id);
```

```bash
warning: variable `__argId` should have a snake case name such as `__arg_id`, #[warn(non_snake_case)] on by default
```

We could massage both the request_uri field to replace `{Id}` with `{id`} and then call `.to_snake_case()` on member_strings on line 107.